### PR TITLE
Move pubsub workflow to EventHubs

### PR DESCRIPTION
**Pull request by halspang**
# Description

This change also includes several improvements to the Pubsub Workflow such as a metric that indicates how long it has been in between events. It also includes a sample of how to setup prometheus scraping in an Azure managed prometheus.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: NA

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: longhaul-test/pubsub-workflow-deploy.yml [hunk 1]
```diff
@@ -42,11 +42,11 @@ spec:
         dapr.io/app-port: "3000"
         dapr.io/log-as-json: "true"
         prometheus.io/scrape: 'true'
-        prometheus.io/port: '9988'  
+        prometheus.io/port: '9988'
     spec:
       containers:
       - name: pubsub-workflow
-        image: daprtests.azurecr.io/pubsub-workflow:dev
+        image: halspang/pubsub-workflow:dev
```
Comment (tmacam) [inline longhaul-test/pubsub-workflow-deploy.yml:49]: Do we really want to refer to your repo?

Comment (halspang) [reply longhaul-test/pubsub-workflow-deploy.yml]: It's a pretty great repo, but no, no we do not :)

**Review by tmacam — APPROVED** [review]

**Comment by berndverst** [comment]
@halspang @tmacam is this PR still relevant?

**Comment by tmacam** [comment]
Not in its totality. There are part of it that are and that were not moved to another PR. I also am not 100% why this move was done and if we want to do something similar in the bicep templates that are still using service bus. I would say that the best is to move it back to draft and wait for Hal to return from his vacation to check if we should split this PR.

In the mean time I can try to move parts of this PR into smaller ones to decrease its size/impact.

**Comment by halspang** [comment]
If I had to guess, this was done because of some specific customer having problems with EventHub. I honestly forget exactly why I did it originally. We can always separate the other improvements into a different PR if we want them, but I don't think we need this anymore.

**Pull request closed without merging**